### ISSUE TITLE
`make` fails with hard to decipher error messages

### ISSUE THREAD

**Issue by yuvipanda** [rocker-org/rocker-versioned2#720]
The title could also be because I'm generally scared of make :)

From the master branch, if I type `make` I get the following error:

```
$ make
jq: error: Could not open file : No such file or directory
[+] Building 0.0s (0/0)                                                                                           
error: open : no such file or directory
jq: error: Could not open file : No such file or directory
jq: error: Could not open file : No such file or directory
make: Nothing to be done for `all'.
```

I'm not sure what is wrong nor how exactly to interpret this.

**Comment by eitsupi** [comment]
I don't think this is a `make`'s failure, just that the default target `all` is a dummy target where nothing actually happens.

https://github.com/rocker-org/rocker-versioned2/blob/485c8bb79bc144d804bd4f0d2ce8f25e0ab5c1e5/Makefile#L5-L6

The `jq` error I have seen many times and am not sure where it is coming from.
Maybe it's because there are a few places where `docker` command are handled by `jq`.

**Comment by yuvipanda** [comment]
Yeah, definitely not an issue with `make` itself - I have updated the title to be clearer that it's *running* `make` that provides this output.

`make setup` also does the same.

And yeah, I too am not sure where it's coming from.

**Comment by yuvipanda** [comment]
I'm also not sure what make target to use to actually locally build all containers.

**Comment by eitsupi** [comment]
In fact, most parts of this Makefile are intended to be used on CI and are not needed locally.

> I'm also not sure what make target to use to actually locally build all containers.

The point here is that `r-ver` is the only file other than Dockerfile that is copied into the container.

https://github.com/rocker-org/rocker-versioned2/blob/485c8bb79bc144d804bd4f0d2ce8f25e0ab5c1e5/dockerfiles/r-ver_devel.Dockerfile#L19

This means that in order to build other images, we must first build `r-ver` and bring the updated rocker_scripts into the container.

In practice, this is too much trouble, so we need to write our own Dockerfile that copies the rocker_scripts into the container, overwrites the rocker_scripts in the container, and builds it.

Like this:

https://github.com/rocker-org/rocker-versioned2/blob/485c8bb79bc144d804bd4f0d2ce8f25e0ab5c1e5/tests/rocker_scripts/Dockerfile#L1-L10

**Comment by eitsupi** [comment]
Honestly, I'm tired of using Make for task runners like this, and this Makefile isn't very good since I wrote it when I knew very little about Make...

I would definitely use [Task](https://taskfile.dev/) instead of Make now.

**Comment by yuvipanda** [comment]
I'm personally a very large proponent of not using Make in general :D

**Comment by eddelbuettel** [comment]
Hm, no, `make` is rock solid.  An [ACM Software Systems Award](https://en.wikipedia.org/wiki/ACM_Software_System_Award) (in 2003, just like S / R in 1998) is rarely wrong.  May I point out (cf #719) that this is a fairly complex project and that using JSON (confession time: I actually think that is not a great choice for config files, apparently I am near alone in that view) in all its glory and dependencies may be related ?  Anyway.

Famous quip about programming languages (by Stroustrup if memory serves) is that there two kinds: the ones people complain about, and the ones nobody uses. Same for build systems. We all have `cmake` too but whatyagottado ....

**Comment by yuvipanda** [comment]
@eddelbuettel would love some help with the error message!

**Comment by eddelbuettel** [comment]
@yuvipanda I hear you but we have a large JSON shaped gap between the two of us!  Those are `jq` errors so ball in your court (or in @eitsupi for the wide selection of R packages).  `make` runs subshells, I find one can generally substitute direct commands.  Would love to help, can't here by toolkit choice.

**Comment by yuvipanda** [comment]
> Would love to help, can't here by toolkit choice.

That makes the two of us, from different sides :)

**Comment by cboettig** [comment]
@eitsupi This error appears to be occurring in current runs of the dockerfiles.yml action too, e.g. https://github.com/rocker-org/rocker-versioned2/actions/runs/6760581560/job/18374524496 .  seems to complain about the runtime env not having `jq`, but then hitting a hard error maybe on parsing `bakefiles/core-latest-daily.docker-bake.json` ?

**Comment by eitsupi** [comment]
> then hitting a hard error maybe on parsing `stacks/devel.json`
That is nothing to do with this issue.
Please see #722
